superset/security/manager.py

```diff
@@ -877,3 +878,34 @@ def assert_viz_permission(self, viz: "BaseViz") -> None:
         """
 
         self.assert_datasource_permission(viz.datasource)
+
+    def get_rls_filters(self, table: "BaseDatasource"):
+        """
+        Retrieves the appropriate row level security filters for the current user and the passed table.
+
+        :param table: The table to check against
+        :returns: A list of filters strings.
+        """
+        if (
+            hasattr(g, "user")
+            and hasattr(g.user, "roles")
+            and hasattr(table, "row_level_security")
+        ):
+            roles = [role.id for role in g.user.roles]
+            return [
+                f
+                for f in table.row_level_security_filters
```

Comment:
Given that by default `sqlalchemy` lazy-loads relationships, this would emit 1+number_of_rls_in_table queries.

Let's write a single `sqlalchemy` query that returns the list of filters with the proper `IN` clause

Reply:
Done - this reduces the number of necessary `hasattr` calls as well.